Sources/Starknet/Data/Transaction/Data/ExecutionResources.swift

```diff
@@ -1,6 +1,42 @@
 import Foundation
 
-public struct StarknetExecutionResources: Decodable, Equatable {
+public protocol StarknetResources: Decodable, Equatable {
+    var steps: Int { get }
+    var memoryHoles: Int? { get }
+    var rangeCheckApplications: Int? { get }
+    var pedersenApplications: Int? { get }
+    var poseidonApplications: Int? { get }
+    var ecOpApplications: Int? { get }
+    var ecdsaApplications: Int? { get }
+    var bitwiseApplications: Int? { get }
+    var keccakApplications: Int? { get }
+}
+
+public struct StarknetComputationResources: StarknetResources {
+    public let steps: Int
+    public let memoryHoles: Int?
+    public let rangeCheckApplications: Int?
+    public let pedersenApplications: Int?
+    public let poseidonApplications: Int?
+    public let ecOpApplications: Int?
+    public let ecdsaApplications: Int?
+    public let bitwiseApplications: Int?
+    public let keccakApplications: Int?
```

Comment:
It seems that [segment_arena_builtin](https://github.com/starkware-libs/starknet-specs/blob/v0.7.0/api/starknet_api_openrpc.json#L3853) is missing.

Reply:
Good catch!